### PR TITLE
Remove configuration for user-defined services SPI

### PR DESCRIPTION
**Pull request by mmedenjak**
In 4.0, we removed support for adding custom services by moving the
NodeEngine and all services to internal packages. With this, we remove
the configuration part for user-defined services. We still allow adding
custom services through a private ConfigAccessor method which is useful
for some tests or for users wanting to use private API at their own risk.

Also did some JDK8 cleanup.

Fixes: https://github.com/hazelcast/hazelcast/issues/15424
EE: https://github.com/hazelcast/hazelcast-enterprise/pull/3334

### CODE REVIEW

File: hazelcast-spring/src/test/resources/com/hazelcast/spring/fullConfig-applicationContext-hazelcast.xml [hunk 1]
```diff
@@ -813,24 +813,6 @@
                 <hz:attribute name="cluster.name">spring-cluster</hz:attribute>
             </hz:member-attributes>
 
-            <hz:services>
```
Comment (vojtechtoman) [inline hazelcast-spring/src/test/resources/com/hazelcast/spring/fullConfig-applicationContext-hazelcast.xml]: The reference to `hazelcast-sample-service.xsd` in the document element should also be removed from here (also from `XmlConfigSchemaLocationTest`), and the file itself should be deleted.

File: hazelcast/src/main/java/com/hazelcast/config/Config.java [hunk 2]
```diff
@@ -2620,27 +2590,61 @@ public Config setMetricsConfig(@Nonnull MetricsConfig metricsConfig) {
...
+    @PrivateApi
+    protected ServicesConfig getServicesConfig() {
+        return servicesConfig;
+    }
+
     @Override
     public String toString() {
         return "Config{"
-                + "clusterName=" + clusterName
+                + "configurationUrl=" + configurationUrl
```
[the remark this answers is not in the document]
Comment (mmedenjak) [reply hazelcast/src/main/java/com/hazelcast/config/Config.java]: You might be right. Don't know. I accidentally noticed some properties were missing so I decided to update it. Maybe I should have just ignored it :)

Seriously, we might use the `ConfigXmlGenerator` to generate the XML as I don't think we have YAML serialization. But I'm reluctant to do anything. I don't like this output as well as not liking outputting XML (it's not expected and this format, as much as it's verbose, is expected and parseable).

[the remark this answers is not in the document]
Comment (vojtechtoman) [reply hazelcast/src/main/java/com/hazelcast/config/Config.java]: Yes, I was thinking the same about the expected format, verbosity etc. Let's pretend this discussion never took place.

File: hazelcast-spring/src/test/resources/com/hazelcast/spring/fullConfig-applicationContext-hazelcast.xml (hunk 1 above)
Comment (mmedenjak) [reply hazelcast-spring/src/test/resources/com/hazelcast/spring/fullConfig-applicationContext-hazelcast.xml]: Removed, please take a look.

File: hazelcast/src/main/java/com/hazelcast/config/Config.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (mmedenjak) [reply hazelcast/src/main/java/com/hazelcast/config/Config.java]: Great, I wanted to suggest opening a GH issue and forgetting about it!

File: hazelcast-spring/src/test/resources/com/hazelcast/spring/fullConfig-applicationContext-hazelcast.xml (hunk 1 above)
Comment (vojtechtoman) [reply hazelcast-spring/src/test/resources/com/hazelcast/spring/fullConfig-applicationContext-hazelcast.xml]: Looking good.

**Comment by mmedenjak** [comment]
Thank you for the reviews!

**Pull request merged**